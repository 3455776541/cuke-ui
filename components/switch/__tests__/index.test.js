import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Switch from "../index";

describe("<Switch/>", () => {
	it("should render a <Switch/> components", () => {
		const wrapper = render(
			<div>
				<Switch />
				<Switch checkedChildren="♂" unCheckedChildren="♀" />
				<Switch checkedChildren="开" unCheckedChildren="关" defaultChecked />
			</div>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should find cuke-switch classnames", () => {
		const wrapper = render(
			<div>
				<Switch />
				<Switch disabled />
				<Switch loading />
				<Switch checkedChildren="♂" unCheckedChildren="♀" />
				<Switch checkedChildren="开" unCheckedChildren="关" defaultChecked />
			</div>
		);
		assert(wrapper.find(".cuke-switch").length >= 1);
		assert(wrapper.find(".cuke-switch-checked").length === 1);
		assert(wrapper.find(".cuke-switch-disabled").length === 1);
		assert(wrapper.find(".cuke-switch-loading").length === 1);
	});

	it("should render sizes", () => {
		const wrapper = render(
			<div>
				<Switch />
				<Switch size="large" />
				<Switch size="small" />
			</div>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
		assert(wrapper.find(".cuke-switch-large").length === 1);
		assert(wrapper.find(".cuke-switch-small").length === 1);
	});

	it("should can trigger click event", () => {
		const onClick = jest.fn();
		const wrapper = shallow(<Switch onChange={onClick} />);
		wrapper.find(".cuke-switch").simulate("click");
		expect(onClick).toHaveBeenCalled();
	});
	it("should can not trigger click event width disabled status", () => {
		const onClick = jest.fn();
		const wrapper = shallow(<Switch onChange={onClick} disabled />);
		wrapper.find(".cuke-switch").simulate("click");
		expect(onClick).not.toHaveBeenCalled();
	});
	it("should can not trigger click event width loading status", () => {
		const onClick = jest.fn();
		const wrapper = shallow(<Switch onChange={onClick} loading />);
		wrapper.find(".cuke-switch").simulate("click");
		expect(onClick).not.toHaveBeenCalled();
	});
});
