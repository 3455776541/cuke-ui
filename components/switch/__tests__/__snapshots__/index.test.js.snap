// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch/> should render a <Switch/> components 1`] = `
<div>
  <span
    class="cuke-switch"
  >
    <span
      class="cuke-switch-inner"
    />
  </span>
  <span
    class="cuke-switch"
  >
    <span
      class="cuke-switch-inner"
    >
      ♀
    </span>
  </span>
  <span
    class="cuke-switch cuke-switch-checked"
  >
    <span
      class="cuke-switch-inner"
    >
      开
    </span>
  </span>
</div>
`;

exports[`<Switch/> should render sizes 1`] = `
<div>
  <span
    class="cuke-switch"
  >
    <span
      class="cuke-switch-inner"
    />
  </span>
  <span
    class="cuke-switch cuke-switch-large"
  >
    <span
      class="cuke-switch-inner"
    />
  </span>
  <span
    class="cuke-switch cuke-switch-small"
  >
    <span
      class="cuke-switch-inner"
    />
  </span>
</div>
`;
